### PR TITLE
Extracted Extractor and Injector code from B3Propagator

### LINKED ISSUE #999: Separate single and combined B3 header logic
The B3Propagator includes logic for both injecting and extacting span context from both single and multiple headers. To help simplify the code, the two code paths could be broken into two classes.

### PR DESCRIPTION
**Pull request by davebarda**
Code polish

### CODE REVIEW

**Comment by jkwatson** [comment]
2 things:
1) Can you update the copyright date on the B3Propagator class?
2) What do you think about splitting up the tests as well?

**Comment by davebarda** [comment]
1. Done
2. I think that testing the public API is better here, as the tests are concise, although the test class is pretty long.

**Comment by carlosalberto** [comment]
IIRC the idea was to have the propagator refactored into two propagators, i.e. single header AND multiple header formats.

From the related B3 effort:

> Should we have 2 propagators: legacy + single-header? I think that simplifies the code. We can do it later.

**Comment by jkwatson** [comment]
Yes, I agree this does not resolve #999, but it is still a good cleanup of the code.

**Comment by carlosalberto** [comment]
So the plan is to end up with, eventually, 4 classes? Single format extractor/injector + Multiple header extractor/injector?

File: contrib/trace_propagators/src/main/java/io/opentelemetry/contrib/trace/propagation/B3PropagatorInjector.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import io.opentelemetry.internal.Utils;
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.SpanContext;
+import io.opentelemetry.trace.SpanId;
+import io.opentelemetry.trace.TraceId;
+import io.opentelemetry.trace.TracingContextUtils;
+import javax.annotation.concurrent.Immutable;
+
+@Immutable
+class B3PropagatorInjector {
```
Comment (bogdandrutu) [inline contrib/trace_propagators/src/main/java/io/opentelemetry/contrib/trace/propagation/B3PropagatorInjector.java:30]: final?

File: contrib/trace_propagators/src/main/java/io/opentelemetry/contrib/trace/propagation/B3PropagatorInjector.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  private static final int SPAN_ID_HEX_SIZE = 2 * SpanId.getSize();
+  private static final int COMBINED_HEADER_DELIMITER_SIZE = 1;
+  private static final int SPAN_ID_OFFSET = TRACE_ID_HEX_SIZE + COMBINED_HEADER_DELIMITER_SIZE;
+  private static final int SAMPLED_FLAG_OFFSET =
+      SPAN_ID_OFFSET + SPAN_ID_HEX_SIZE + COMBINED_HEADER_DELIMITER_SIZE;
+  private static final int COMBINED_HEADER_SIZE = SAMPLED_FLAG_OFFSET + SAMPLED_FLAG_SIZE;
+
+  private final boolean singleHeader;
+
+  public B3PropagatorInjector(boolean singleHeader) {
```
Comment (bogdandrutu) [inline contrib/trace_propagators/src/main/java/io/opentelemetry/contrib/trace/propagation/B3PropagatorInjector.java:42]: remove public?

File: contrib/trace_propagators/src/main/java/io/opentelemetry/contrib/trace/propagation/B3PropagatorInjector.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    this.singleHeader = singleHeader;
+  }
+
+  public <C> void inject(Context context, C carrier, HttpTextFormat.Setter<C> setter) {
```
Comment (bogdandrutu) [inline contrib/trace_propagators/src/main/java/io/opentelemetry/contrib/trace/propagation/B3PropagatorInjector.java:46]: same, class is package protected, no need to be public.

File: contrib/trace_propagators/src/main/java/io/opentelemetry/contrib/trace/propagation/B3PropagatorExtractor.java [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+import io.opentelemetry.trace.TraceFlags;
+import io.opentelemetry.trace.TraceId;
+import io.opentelemetry.trace.TraceState;
+import io.opentelemetry.trace.TracingContextUtils;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.concurrent.Immutable;
+
+@Immutable
+class B3PropagatorExtractor {
```
Comment (bogdandrutu) [inline contrib/trace_propagators/src/main/java/io/opentelemetry/contrib/trace/propagation/B3PropagatorExtractor.java:46]: final?

File: contrib/trace_propagators/src/main/java/io/opentelemetry/contrib/trace/propagation/B3PropagatorExtractor.java [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+  private static final Logger logger = Logger.getLogger(B3PropagatorExtractor.class.getName());
+
+  private final boolean isSingleHeader;
+
+  public B3PropagatorExtractor(boolean isSingleHeader) {
```
Comment (bogdandrutu) [inline contrib/trace_propagators/src/main/java/io/opentelemetry/contrib/trace/propagation/B3PropagatorExtractor.java:51]: No public, class is package protected.

File: contrib/trace_propagators/src/main/java/io/opentelemetry/contrib/trace/propagation/B3PropagatorExtractor.java [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+    this.isSingleHeader = isSingleHeader;
+  }
+
+  public <C> Context extract(Context context, C carrier, HttpTextFormat.Getter<C> getter) {
```
Comment (bogdandrutu) [inline contrib/trace_propagators/src/main/java/io/opentelemetry/contrib/trace/propagation/B3PropagatorExtractor.java:55]: same.

File: contrib/trace_propagators/src/main/java/io/opentelemetry/contrib/trace/propagation/B3Propagator.java [hunk 7]
```diff
@@ -42,40 +32,27 @@
...
   static final String SPAN_ID_HEADER = "X-B3-SpanId";
   static final String SAMPLED_HEADER = "X-B3-Sampled";
   static final String TRUE_INT = "1";
   static final String FALSE_INT = "0";
   static final String COMBINED_HEADER = "b3";
   static final String COMBINED_HEADER_DELIMITER = "-";
+  static final int MAX_TRACE_ID_LENGTH = 2 * TraceId.getSize();
+  static final int MAX_SPAN_ID_LENGTH = 2 * SpanId.getSize();
+  static final TraceFlags SAMPLED_FLAGS = TraceFlags.builder().setIsSampled(true).build();
+  static final TraceFlags NOT_SAMPLED_FLAGS = TraceFlags.builder().setIsSampled(false).build();
```
Comment (bogdandrutu) [inline contrib/trace_propagators/src/main/java/io/opentelemetry/contrib/trace/propagation/B3Propagator.java:45]: Can be moved to extract and private static.

**Comment by davebarda** [comment]
> IIRC the idea was to have the propagator refactored into two propagators, i.e. single header AND multiple header formats.
> 
> From the related B3 effort:
> 
> > Should we have 2 propagators: legacy + single-header? I think that simplifies the code. We can do it later.

Totally correct, I'll try to implement it in a different PR

File: contrib/trace_propagators/src/main/java/io/opentelemetry/contrib/trace/propagation/B3PropagatorInjector.java [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+  private final boolean singleHeader;
+
+  B3PropagatorInjector(boolean singleHeader) {
+    this.singleHeader = singleHeader;
+  }
+
+  <C> void inject(Context context, C carrier, HttpTextFormat.Setter<C> setter) {
+    Utils.checkNotNull(context, "context");
+    Utils.checkNotNull(setter, "setter");
+
```
Comment (bogdandrutu) [inline contrib/trace_propagators/src/main/java/io/opentelemetry/contrib/trace/propagation/B3PropagatorInjector.java:49]: Objects.requireNotNull() and everywhere.

Comment (davebarda) [reply contrib/trace_propagators/src/main/java/io/opentelemetry/contrib/trace/propagation/B3PropagatorInjector.java]: Done, anyway i'll add another PR that removes some non native implementations.

**Comment by bogdandrutu** [comment]
Please remove the extra file.

File: .gradletasknamecache [hunk 9]
```diff
@@ -0,0 +1,2574 @@
...
+verifyGoogleJavaFormat
+verifyGoogleJavaFormat
+verifyGoogleJavaFormat
+verifyGoogleJavaFormat
+verifyGoogleJavaFormat
+verifyGoogleJavaFormat
+verifyGoogleJavaFormat
+verifyGoogleJavaFormat
+verifyGoogleJavaFormat
+verifyGoogleJavaFormat
```
Comment (jkwatson) [inline .gradletasknamecache:2574]: please remove this file, as @bogdandrutu said.

Comment (davebarda) [reply .gradletasknamecache]: Done, why not to put it in git ignore?

Comment (bogdandrutu) [reply .gradletasknamecache]: If you want to do that, please make a separate PR. But we definitely don't want to merge that file :)

**Pull request merged**